marshal/claim.go

```diff
@@ -501,9 +501,13 @@ func (c *claim) healthCheck() bool {
 		return true
 	}
 
-	// If the consumer is moving faster than the partition, consider it healthy. This case
-	// is true in the standard catching up from behind case.
-	if partitionVelocity <= consumerVelocity {
+	// At this point we know the consumer is NOT PRESENTLY caught up or predicted to catch
+	// up in the next two heartbeats.
+
+	// If the consumer is moving as fast or faster than the partition and the consumer is
+	// at least moving, consider it healthy. This case is true in the standard catching
+	// up from behind case.
+	if partitionVelocity < consumerVelocity || (partitionVelocity <= 0 && consumerVelocity > 0) {
```

Comment:
I think one of us is confused but I don't know who.

I am reading the if statement as redundant. The left hand side is `p < c` and the second half is `p non-positive, c positive`. But the former implies the later, so `p < c` is the exact same if statement. Unless I'm the confused one. :)